dokku/deploy_dokku.sh

```diff
@@ -0,0 +1,249 @@
+#!/bin/bash
+set -euo pipefail
+
+# ─────────────────────────────────────────────────────────────
+# Script: deploy_dokku_stack.sh
+# Usage: ./deploy_dokku_stack.sh <create|delete> <LOCATION> <PROJECT_NAME>
+# Requires: upctl, kubectl, helm, jq, make, ssh keys, github PAT env vars
+# ─────────────────────────────────────────────────────────────
+
+export VERSION=1.0.0
+log() { echo "[INFO] $1"; }
+error_exit() { echo "[ERROR] $1" >&2; exit 1; }
+
+# Parse args
+if [[ $# -ne 3 ]]; then
+  echo "Usage: $0 <deploy|delete> <LOCATION> <PROJECT_NAME>"
+  exit 1
+fi
+
+ACTION="$1"
+LOCATION="$2"
+PROJECT="$3"
+export CLUSTER_NAME="dokku-${PROJECT}-${LOCATION}"
+export PRIVATE_NETWORK_NAME="dokku-net-${PROJECT}-${LOCATION}"
+export KUBECONFIG_FILE="/tmp/${CLUSTER_NAME}_kubeconfig.yaml"
+
+# Config values
+export CERT_MANAGER_EMAIL="${CERT_MANAGER_EMAIL:-ops@example.com}"
+export SSH_PATH="${SSH_PATH:-$HOME/.ssh/id_rsa}"
+export SSH_PUB_PATH="${SSH_PUB_PATH:-${SSH_PATH}.pub}"
+export GITHUB_PACKAGE_URL="${GITHUB_PACKAGE_URL:-ghcr.io}"
+GLOBAL_DOMAIN="${GLOBAL_DOMAIN:-}"
+export NUM_NODES="${NUM_NODES:-1}"
+
+check_or_create_network() {
+  NETWORK_ID=$(upctl network list -o json |
+    jq -r ".networks[] | select(.name==\"$PRIVATE_NETWORK_NAME\" and .zone==\"$LOCATION\").uuid")
```

Comment:
To prevent jq from failing if there are no networks:

```suggestion
    jq -r ".networks // [] | .[] | select(.name==\"$PRIVATE_NETWORK_NAME\" and .zone==\"$LOCATION\").uuid")
```